lib/elastic_apm/central_config.rb

```diff
@@ -129,9 +129,13 @@ def handle_error(error)
         error.response.body, DEFAULT_MAX_AGE
       )
 
+    ensure
       assign({})
-
-      schedule_next_fetch(error.response)
+      if error.respond_to?(:response)
+        schedule_next_fetch(error.response)
+      else
+        schedule_next_fetch
+      end
     end
```

Comment:
I feel this part is a bit weird. We're rescue'ing with `ensure` from calling `.response` on something not having a `.response` and then we do the `.respond_to` dance afterwards.

How about:

```ruby
    def handle_error(error)
      # For tests, WebMock failures don't have real responses
      response = error.respond_to(:response) ? error.response : nil

      debug(
        'Failed fetching config: %s, trying again in %d seconds',
        response&.body, DEFAULT_MAX_AGE
      )

      assign({})

      schedule_next_fetch(response)
    end
```

Reply:
I had the `ensure` in there because I was concerned that this method throwing an uncaught error could have such a significant effect on the test suite. It's not entirely necessary to have it but if we see strange "bleeding" behavior in the test suite, it's worth checking this--and similar-- methods for uncaught exceptions.

Reply:
We don’t see them because the agent swallows and logs most of them. We talked about having `Logging#error` raise if test. I think that is a good approach to try?

Reply:
Yes, but my concern is not that errors are swallowed. I think it's a good approach to raise in tests from Logging#error.
My concern was that I observed that following some errors in this method, the agent wasn't able to stop the central config tasks and the agent was still running in other tests. I'll try to find a build to show you, it was sometime last week.

Reply:
Unfortunately the build from last week is no longer available in Jenkins. In any case, let's take the approach of raising in `Logging#error`.